identity-mdoc/src/androidMain/kotlin/com/android/identity/mdoc/transport/BleCentralManagerAndroid.kt

```diff
@@ -0,0 +1,738 @@
+                    throw error
+                }
+            } catch (error: Throwable) {
+                throw error
+            }
+        }
+    }
+
+    override suspend fun requestMtu() {
+        check(device != null && gatt != null)
+        suspendCancellableCoroutine<Boolean> { continuation ->
+            setWaitCondition(WaitState.REQUEST_MTU, continuation)
+            gatt!!.requestMtu(515)
+        }
+    }
+
+    override suspend fun peripheralDiscoverServices(uuid: UUID) {
+        check(device != null && gatt != null)
+        suspendCancellableCoroutine<Boolean> { continuation ->
+            setWaitCondition(WaitState.PERIPHERAL_DISCOVER_SERVICES, continuation)
+            gatt!!.discoverServices()
+        }
+        service = gatt!!.getService(uuid.toJavaUuid())
+        if (service == null) {
+            throw Error("No service with the given UUID")
+        }
+    }
+
+    override suspend fun peripheralDiscoverCharacteristics() {
+        check(device != null && gatt != null && service != null)
+        characteristicState = service!!.getCharacteristic(stateCharacteristicUuid.toJavaUuid())
+        if (characteristicState == null) {
+            throw Error("State characteristic not found")
+        }
+        characteristicClient2Server =
+            service!!.getCharacteristic(client2ServerCharacteristicUuid.toJavaUuid())
+        if (characteristicClient2Server == null) {
+            throw Error("Client2Server characteristic not found")
+        }
+        characteristicServer2Client =
+            service!!.getCharacteristic(server2ClientCharacteristicUuid.toJavaUuid())
+        if (characteristicServer2Client == null) {
+            throw Error("Server2Client characteristic not found")
+        }
+        if (identCharacteristicUuid != null) {
+            characteristicIdent = service!!.getCharacteristic(identCharacteristicUuid!!.toJavaUuid())
+            if (characteristicIdent == null) {
+                throw Error("Ident characteristic not found")
+            }
+        }
+        if (l2capCharacteristicUuid != null) {
+            characteristicL2cap = service!!.getCharacteristic(l2capCharacteristicUuid!!.toJavaUuid())
+            if (characteristicL2cap == null) {
+                Logger.i(TAG, "L2CAP characteristic requested but not found")
+            } else {
+                suspendCancellableCoroutine<Boolean> { continuation ->
+                    setWaitCondition(WaitState.GET_L2CAP_PSM, continuation)
+                    gatt!!.readCharacteristic(characteristicL2cap)
+                }
+            }
+        }
+    }
+
+    var expectedIdentValue: ByteArray? = null
+
+    override suspend fun checkReaderIdentMatches(eSenderKey: EcPublicKey) {
+        check(device != null && gatt != null && identCharacteristicUuid != null)
+
+        val ikm = Cbor.encode(Tagged(24, Bstr(Cbor.encode(eSenderKey.toCoseKey().toDataItem()))))
```

Comment:
Yes, this could be using `Tagged.ENCODED_CBOR` instead of the literal 24... I generally prefer using the literal 24 b/c it's such an ingrained thing with CBOR (it appears in CDDL) and so on. It's almost the same as you'd want a string to be `"here's some text\nhere's the second line"` and not `"here's some text" + Constants.NEW_LINE + "here's the second line"` ... so I'm a little apprehensive to change it 😄